### PR TITLE
lib/model: Use up to date device name, do not provide name to unknown devices (fixes #4164)

### LINKED ISSUE #4164: Changed system name requires restart to take effect in outgoing hello messages
Synchthing ignores user Device Name change and advertises user System Name. A Security Risk

The synchthing user can change their device name from what appears to be the coding pulling that directly from the system platform name. However the system platform name is still shown to all remote peers your ID is shared with. That's a real problem. 

Urgent suggestion, when the synchthing user changes the Device Name in the Webui, that is the Device Name that shows to any and all remote peers you share your ID with downstream.

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
### Purpose

As per ticket.

### Testing

Em, it compiles.

### CODE REVIEW

File: lib/model/model.go [hunk 1]
```diff
@@ -1320,9 +1318,13 @@ func (m *Model) OnHello(remoteID protocol.DeviceID, addr net.Addr, hello protoco
 }
 
 // GetHello is called when we are about to connect to some remote device.
-func (m *Model) GetHello(protocol.DeviceID) protocol.HelloIntf {
+func (m *Model) GetHello(id protocol.DeviceID) protocol.HelloIntf {
+	name := ""
+	if _, ok := m.cfg.Device(id); ok {
+		name = m.cfg.MyName()
```
Comment (calmh) [inline lib/model/model.go]: Use the device you got from m.cfg.Device() here and there is no point in having the cfg.MyName method any more at all?

Comment (AudriusButkevicius) [inline lib/model/model.go]: The ID is the remote ID in this case

Comment (calmh) [inline lib/model/model.go]: Oh. Right.

File: lib/config/wrapper.go [hunk 2]
```diff
@@ -431,3 +431,15 @@ func (w *Wrapper) StunServers() []string {
 
 	return addresses
 }
+
+func (w *Wrapper) MyName() string {
+	w.mut.Lock()
+	myID := w.cfg.MyID
+	w.mut.Unlock()
+	device, ok := w.Device(myID)
```
Comment (calmh) [inline lib/config/wrapper.go]: I think just skip the comma0ok and return device.Name regardless. It should /always/ exist, and if it doesn't it's something really weird with the config that we'll fix on next startup so no need to tell the user something they can't do anything about.

**Comment by calmh** [comment]
Otherwise all lgtm good stuff

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `failure` -- refusing to merge.

**Comment by calmh** [comment]
dde34b13fa7f0de5421cca8888ff29b1_nelson-ha-ha-meme-ha-memes_400-373

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by st-review** [comment]
:ok_hand: Merged as 91d37f35bc7dc6bd08999e85c9d91d54c48fab32. Thanks, @AudriusButkevicius!

**Pull request closed without merging**